### PR TITLE
Integrate changes in the API

### PR DESCRIPTION
**Pull request by wreise**
With the new release, the API of mapper has changed. This pull requests attempts to make it compatible.

Issues:
1. The colors/colormaps are not exactly the same,
2. The requirements still reference the old version of giotto.

### CODE REVIEW

**Comment by wreise** [comment]
@ulupo could you please review and merge this?

File: utils.py [hunk 1]
```diff
@@ -6,6 +6,23 @@
...
+    """Copy v and put NaNs in indices
+    Parameters
+    ----------
+    v : np array of values,
+
+    indices : list or iterable of indices, for whch the value should be set to NaN
+
+    Returns
+    -------
+    v_  : acopy of v, with NaNs at inidices given by `indices`
```
Comment (ulupo) [inline utils.py]: Correct "acopy:

Comment (ulupo) [reply utils.py]: And "inidices"

File: plotting.py [hunk 2]
```diff
@@ -2,11 +2,11 @@
 import utils
 import itertools
 import collections
-from giotto.mapper import visualization
+from gtda.mapper import visualization
```
Comment (ulupo) [inline plotting.py]: I'd prefer "from gtda.mapper import plot_static_mapper_graph" (plus any other if needed, e.g. the interactive one)

File: plotting.py [hunk 3]
```diff
@@ -41,28 +41,29 @@ def get_region_plot(graph, data, layout, columns_to_color, node_elements,
...
-        'node_trace_marker_showscale': False}
-
-    return visualization.create_network_2d(graph, data, layout, node_color,
-                                           columns_to_color=columns_to_color,
-                                           plotly_kwargs=plotly_kwargs)
+        'node_trace_marker_showscale': False,
+        'node_trace_hoverlabel': node_color,
+        'node_trace_marker_color': node_color
+    }
+    return visualization.plot_static_mapper_graph(pipe, data,
```
Comment (ulupo) [inline plotting.py]: Remove "visualization." if the import statement is changed.

File: plotting.py [hunk 4]
```diff
@@ -121,9 +119,11 @@ def get_graph_plot_colored_by_election_results(graph, year, df, data,
                                       df[df['year'] == year]['n_electors']
                                       .reset_index(drop=True)))}
 
-    return visualization.create_network_2d(graph, data, layout, node_color,
-                                           columns_to_color=columns_to_color,
-                                           plotly_kwargs=plotly_kwargs)
+    return visualization.plot_static_mapper_graph(pipeline, data,
```
Comment (ulupo) [inline plotting.py]: Ditto as above.

File: utils.py [hunk 5]
```diff
@@ -6,6 +6,23 @@
 import pandas as pd
 
 
+def get_filtered_values(v, indices):
+    """Copy v and put NaNs in indices
+    Parameters
+    ----------
+    v : np array of values,
+
+    indices : list or iterable of indices, for whch the value should be set to NaN
```
Comment (ulupo) [inline utils.py]: Correct "whch"

File: utils.py [hunk 6]
```diff
@@ -6,6 +6,23 @@
 import pandas as pd
 
 
+def get_filtered_values(v, indices):
+    """Copy v and put NaNs in indices
+    Parameters
+    ----------
+    v : np array of values,
```
Comment (ulupo) [inline utils.py]: "np array" should be "ndarray" (following scikit-learn convention)

Comment (ulupo) [reply utils.py]: Remove hanging comma

File: plotting.py (hunk 2 above)
Comment (wreise) [reply plotting.py]: I applied the changes to the notebook as well.

**Comment by wreise** [comment]
Perhaps we want to merge into master, instead of the `dev` branch?

**Comment by ulupo** [comment]
I'm not 100% aware of the reasons for this dev branch, perhaps @WeilerP or @lewtun remember?

**Comment by lewtun** [comment]
@wreise can you please add the seaborn import and rerun the notebook so @WeilerP can inspect the output?

Screen Shot 2020-02-07 at 17 02 01

**Comment by wreise** [comment]
Sorry for the delay @WeilerP !

**Comment by wreise** [comment]
@matsantoro, once [PR](https://github.com/giotto-ai/giotto-tda/pull/277) is merged,  I will merge this one, which should fix #8, #9, #10 .

**Comment by wreise** [comment]
Even after the merge of [#277](https://github.com/giotto-ai/giotto-tda/pull/277), there are still problems:
1. colors in the `Colored by Economic Indicator` section are all purple, if we do not touch the dropdown list. In addition, the isolated CCs should be somehow excluded.
2. In the graph above the map, the colors should be the same as those on that map, and 'cluster well'.
3. In `Color by Presidential Election`, the graph should be the same (based on 2016 elections), and only the colors should be different.

**Comment by wreise** [comment]
It appears we need [#286](https://github.com/giotto-ai/giotto-tda/pull/286).

**Comment by wreise** [comment]
@ulupo , this should be all  good now, and reproducing the results from the [post](https://towardsdatascience.com/the-shape-of-the-united-states-presidential-elections-c336d80e4ddf).

**Comment by ulupo** [comment]
Thanks @wreise! I guess it is worth remembering that at this point it will only work with the dev installation. Correct?

**Comment by wreise** [comment]
This is exactly what I realized- I updated the requirements. Once it is out, i will change them again   to 20200214

**Comment by ulupo** [comment]
Perfect, shall we wait until then to merge to have a convenient reminder?

**Comment by wreise** [comment]
The nightly seems  to be on pypi - thanks!

**Comment by lewtun** [comment]
@wreise and @ulupo should we merge this?

**Comment by wreise** [comment]
@ulupo please have a look and merge if appropriate (i don't have write access)

**Comment by ulupo** [comment]
Sorry, this escaped my attention, let's do it!

**Comment by wreise** [comment]
aj! I put the target branch as `dev` instead of `master`. Could you merge it to master, please?

**Pull request merged**